### PR TITLE
Support Neo4j 5.x

### PR DESCRIPTION
**Pull request by ianmkenney**
This PR allows to use of Neo4j 5.x with `alchemiscale`.

### CODE REVIEW

**Comment by ianmkenney** [comment]
To get this working, `grolt` needs some tweaking. I couldn't find a path to update the relevant configuration changes between Neo4j 4.x and 5.x.

**Comment by ianmkenney** [comment]
This is ready for review after #235 is merged and we rebase this branch against `main`

**Comment by dotsdl** [comment]
@ianmkenney reviewing now!

File: alchemiscale/storage/statestore.py [hunk 1]
```diff
@@ -235,11 +231,7 @@ def _store_check(self):
 
     def reset(self):
         """Remove all data from database; undo all components in `initialize`."""
-        # we'll keep NOPE around to avoid a case where Neo4j doesn't give it id 0
-        # after a series of wipes; appears to happen often enough in tests
-        # can remove this once py2neo#951 merged
-        # self.graph.run("MATCH (n) DETACH DELETE n")
-        self.execute_query("MATCH (n) WHERE NOT n:NOPE DETACH DELETE n")
+        self.graph.execute_query("MATCH (n) DETACH DELETE n")
```
Comment (dotsdl) [inline alchemiscale/storage/statestore.py:234]:
```suggestion
        self.execute_query("MATCH (n) DETACH DELETE n")
```

**Review by dotsdl — APPROVED** [review]
Just one edit needed given changes in #235. Otherwise looks great @ianmkenney!

**Pull request merged**